packages/amplify-category-api/commands/api/add-graphql-datasource.js

```diff
@@ -205,3 +207,42 @@ async function getAwsClient(context, action) {
   const provider = require(providerPlugins[providerName]);
   return await provider.getConfiguredAWSClient(context, 'aurora-serverless', action);
 }
+
+function readSchema(projectDirectory, rdsSchema) {
+  const schemaFilePath = path.join(projectDirectory, 'schema.graphql');
+  const schemaDirectoryPath = path.join(projectDirectory, 'schema');
+  const schemaDirectoryExists = fs.existsSync(schemaDirectoryPath);
+  // convert the object schema to dump as file
+  const rdsSchemaDoc = mergeTypes([{}, rdsSchema], { all: true });
+
+  let schema;
+  if (schemaDirectoryExists) {
+    schema = readSchemaDocuments(schemaDirectoryPath).join('\n');
+    const rdsSchemaFilePath = path.join(schemaDirectoryPath, 'rds.graphql');
+    fs.writeFileSync(rdsSchemaFilePath, rdsSchemaDoc, 'utf8');
+  } else {
+    schema = fs.readFileSync(schemaFilePath).toString();
+  }
+  return schema;
+}
+
+function readSchemaDocuments(schemaDirectoryPath) {
+  const files = fs.readdirSync(schemaDirectoryPath);
+  let schemaDocuments = [];
+  for (const fileName of files) {
+    if (fileName.indexOf('.') === 0) {
+      continue;
+    }
+
+    const fullPath = `${schemaDirectoryPath}/${fileName}`;
+    const stats = fs.lstatSync(fullPath);
+    if (stats.isDirectory()) {
+      const childDocs = readSchemaDocuments(fullPath);
```

Comment:
I don't think graphql-transformer supports sub-directories inside schema folder. Could you verify this and keep this code if and only if it is supported

Reply:
Please correct me if I am wrong here. My understanding we are gathering all the schema docs and generating a single schema.graphql file. Then graphql transformer will read this file to generate resolvers and e.t.c

How is the schema directory dependent on graphql transformer?

Reply:
GraphQL transformer has support for merging schemas inside `schema` directory. Otherwise what would be the need for checking for existence of schema directory and merging?
Relevant code:
https://github.com/aws-amplify/amplify-cli/blob/bcabc528dc79ebfcd128b7e216e889dc61dd38be/packages/graphql-transformer-core/src/util/transformConfig.ts#L246-L260

Reply:
okay got it. I misunderstood the first comment. You were saying to check if the graphql transformer also supports this functionality.